web-client/src/modules/create/containers/CreatePostContainer.tsx

```diff
@@ -1,336 +1,196 @@
 
-import NewPost from '../components/NewPost';
-import PostConfirmation from '../components/PostConfirmation';
-import PostReview, { RequestInput } from '../components/PostReview';
-import { CreatePostTypes } from '../constants';
-
-/* TODO:  integrate with translation if safe */
-const DELIVERIES = 'Deliveries';
-
-// TS interfaces for state later on.
-
-// enum Type {
-//   delivery = 'Deliveries',
-//   other = 'Other',
-// }
-
-// interface RequestDetails {
-//   title: string;
-//   type: Type;
-//   details: string;
-// }
-
-// interface RequestAddress {
-//   streetAddress: string;
-//   latLng: firestore.GeoPoint;
-// }
-
-const CreatePostContainer: React.FC<ICreatePostContainer> = () => {
-  const { t } = useTranslation();
+const CreatePostContainer: React.FC = () => {
+  const dispatch = useDispatch();
   const history = useHistory();
+  const { t } = useTranslation();
 
-  const [requestInfo, setRequestInfo] = useState<RequestInput | undefined>(
-    undefined,
-  );
-
-  const [isSubmitting, setIsSubmitting] = useState<boolean>(false);
-
-  const [showReviewPage, setShowReviewPage] = useState<boolean>(false);
-
-  const [showConfirmationPage, setShowConfirmationPage] = useState<boolean>(
-    false,
+  const profileState = useSelector(
+    ({ profile }: { profile: ProfileState }) => profile,
   );
-
-  const [authModalIsVisible, setAuthModalIsVisible] = useState<boolean>(false);
-
-  const onboarded = useSelector((state: AppState) => state.auth.onboarded);
-
   const phoneNumber = useSelector(
     (state: AppState) => state.auth.user?.phoneNumber,
   );
+  const onboarded = useSelector((state: AppState) => state.auth.onboarded);
 
-  const profileState = useSelector(
-    ({ profile }: { profile: ProfileState }) => profile,
-  );
-
-  // *** new state to be implemented later on ***
-
-  // const [requestDetails, setRequestDetails] = useState<RequestDetails>({
-  //   title: '',
-  //   type: Type.delivery,
-  //   details: '',
-  // });
-
-  // const coordsFromProfile = getCoordsFromProfile(profileState);
-
-  // const [requestAddress, setRequestAddress] = useState<RequestAddress>({
-  //   streetAddress: getStreetAddressFromProfile(profileState) || 'Address could not be found',
-  //   latLng: new firestore.GeoPoint(coordsFromProfile.lat, coordsFromProfile.lng),
-  // });
-
-  const [mapAddress, setMapAddress] = useState<string>(
-    () =>
-      getStreetAddressFromProfile(profileState) || 'Address could not be found',
-  );
+  /* steps */
+  const [stepNumber, setStepNumber] = useState(0);
+  const moveForwards = () => setStepNumber(stepNumber + 1);
+  const moveBackwards = () => setStepNumber(stepNumber - 1);
+
+  /* PostDetails */
+  const getTypes = () => [t('modules.create.defaults.postDetails.type')];
+  const [postDetails, setPostDetails] = useState<IPostDetails>({
+    title: t('modules.create.defaults.postDetails.title'),
+    body: t('modules.create.defaults.postDetails.body'),
+    type: getTypes()[0],
```

Comment:
Looks great to me overall functionality wise, but hardcoding default values might cause people to miss filling them out themselves. Instead probably best to keep these as an example `placeholder` for each `Input` component in `PostDetailsStep.tsx`

Reply:
fixed